kernel/hardware/miniuart.cpp

```diff
@@ -0,0 +1,110 @@
+#include <cstddef>
+#include <cstdint>
+
+#include "gpio.hpp"
+#include "miniuart.hpp"
+#include "mmio.hpp"
+
+namespace MINI_UART {
+/** Auxiliary Interrupt status (size: 3) */
+// static constexpr int32_t AUX_IRQ = 0x215000;
+
+/** Auxiliary enables (size: 3) */
+static constexpr int32_t AUX_ENABLES = 0x215004;
+
+/** Mini UART I/O Data (size: 8) */
+static constexpr int32_t AUX_MU_IO_REG = 0x215040;
+
+/** Mini UART Interrupt Enable (size: 8) */
+static constexpr int32_t AUX_MU_IER_REG = 0x215044;
+
+/** Mini UART Interrupt Identify (size: 8) */
+// static constexpr int32_t AUX_MU_IIR_REG = 0x215048;
+
+/** Mini UART Line Control (size: 8) */
+static constexpr int32_t AUX_MU_LCR_REG = 0x21504C;
+
+/** Mini UART Modem Control (size: 8) */
+static constexpr int32_t AUX_MU_MCR_REG = 0x215050;
+
+/** Mini UART Line Status (size: 8) */
+static constexpr int32_t AUX_MU_LSR_REG = 0x215054;
+
+/** Mini UART Extra Control (size: 8) */
+static constexpr int32_t AUX_MU_CNTL_REG = 0x215060;
+
+/** Mini UART Baud rate (size: 16) */
+static constexpr int32_t AUX_MU_BAUD_REG = 0x215068;
+
+#if RASPI_VERSION == 3
+static constexpr uint64_t CORE_CLOCK = 250'000'000;
+#elif RASPI_VERSION == 4
+static constexpr uint64_t CORE_CLOCK = 200'000'000;
+#endif
+
+void init(uint32_t baud_rate) {
+  // We deactivate Pull Up/Down fot the pins 14 and 15
```

Comment:
Can we choose any `baud_rate`? If no, I think it will be better to either add a `KASSERT` or check at runtime if the rate is supported (and emit an error log after a fallback to a default value).

Do you agree?

Reply:
I have tested with multiplies Band Rate, but I haven't checked with non standard ones !